python/cuspatial/cuspatial/core/binpreds/binpreds.py

```diff
@@ -306,63 +305,108 @@ def _offset_equals(self, lhs, rhs):
         rhs_lengths = rhs[:-1] - rhs[1:]
         return lhs_lengths == rhs_lengths
 
-    def _sort_multipoints(self, lhs, rhs, initial):
+    def _sort_interleaved_points_by_offset(self, coords, offsets, sort_order):
+        """Sort xy according to bins defined by offset. Sort order is a list
+        of column names to sort by.
+
+        `_sort_interleaved_points_by_offset` creates a dataframe with the
+        following columns:
+        "sizes": an index for each object represented in `coords`.
+        "points": an index for each point in `coords`.
+        "xy_key": an index that maintains x/y ordering.
+        "xy": the x/y coordinates in `coords`.
+
+        The dataframe is sorted according to keys passed in by the caller.
+        For sorting multipoints, the keys in order are "object_key", "xy",
+        "xy_key". This sorts the points in each multipoint into the same
+        bin defined by "object_key", then sorts the points in each bin by
+        x/y coordinates, and finally sorts the points in each bin by x/y
+        ordering.
```

Comment:
> x/y ordering

Just checking, I assume you mean x coordinate should precede y coordinate?

Reply:
Yes, that's the purpose of the `xy_key`.